clang-tools-extra/clang-doc/HTMLGenerator.cpp

```diff
@@ -490,15 +490,15 @@ genReferencesBlock(const std::vector<Reference> &References,
   }
   return Out;
 }
-
 static std::unique_ptr<TagNode>
-writeFileDefinition(const Location &L,
-                    std::optional<StringRef> RepositoryUrl = std::nullopt) {
-  if (!L.IsFileInRootDir && !RepositoryUrl)
+writeFileDefinition(const ClangDocContext &CDCtx, const Location &L) {
```

Comment:
Lets change this to `writeSourceFileRef()` and your helper can become `maybeWriteSourceFileRef()`. I think those are quite a bit more accurate names.

`writeDefLoc()` and `maybeWriteDefLoc()` are also potentially better.